### PR TITLE
Features rot90

### PR DESCRIPTION
**Pull request by mtar**
## Description

Issue/s resolved: #555 

Introduces rot90. Rotate an array on a specified plane. May change the split axis if it is in the rotation plane similar to transpose.

## Changes proposed:
- add rot90()

## Type of change

New feature

## Due Diligence

- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/core/manipulations.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from . import stride_tricks
 from . import tiling
 from . import types
+from . import linalg
```
Comment (Markus-Goetz) [inline heat/core/manipulations.py:12]: please sort in alphabetically

File: heat/core/manipulations.py [hunk 2]
```diff
@@ -936,6 +938,86 @@ def reshape_argsort_counts_displs(
     return factories.array(data, dtype=a.dtype, is_split=axis, device=a.device, comm=a.comm)
 
 
+def rot90(m, k=1, axes=(0, 1)):
```
Comment (Markus-Goetz) [inline heat/core/manipulations.py:941]: please use return type annotation

File: heat/core/manipulations.py [hunk 3]
```diff
@@ -936,6 +938,86 @@ def reshape_argsort_counts_displs(
...
+    ----------
+    m : DNDarray
+        Array of two or more dimensions.
+    k : integer
+        Number of times the array is rotated by 90 degrees.
+    axes: (2,) int list or tuple
+        The array is rotated in the plane defined by the axes.
+        Axes must be different.
+
+    Returns
```
Comment (Markus-Goetz) [inline heat/core/manipulations.py:955]: Remove this section

File: heat/core/manipulations.py [hunk 4]
```diff
@@ -936,6 +938,86 @@ def reshape_argsort_counts_displs(
     return factories.array(data, dtype=a.dtype, is_split=axis, device=a.device, comm=a.comm)
 
 
+def rot90(m, k=1, axes=(0, 1)):
+    """
+    Rotate an array by 90 degrees in the plane specified by axes.
+    Rotation direction is from the first towards the second axis.
+    Parameters
```
Comment (Markus-Goetz) [inline heat/core/manipulations.py:945]: newline

File: heat/core/manipulations.py [hunk 5]
```diff
@@ -936,6 +938,86 @@ def reshape_argsort_counts_displs(
...
+
+    if axes[0] >= m.ndim or axes[0] < -m.ndim or axes[1] >= m.ndim or axes[1] < -m.ndim:
+        raise ValueError("Axes={} out of range for array of ndim={}.".format(axes, m.ndim))
+
+    if m.split is None:
+        return factories.array(
+            torch.rot90(m._DNDarray__array, k, axes), dtype=m.dtype, device=m.device, comm=m.comm
+        )
+
+    k %= 4
```
Comment (Markus-Goetz) [inline heat/core/manipulations.py:1004]: try-except block for ``k = int(k)``

File: heat/core/manipulations.py [hunk 6]
```diff
@@ -936,6 +938,86 @@ def reshape_argsort_counts_displs(
...
+
+    if k == 0:
+        return m[:]
```
Comment (Markus-Goetz) [inline heat/core/manipulations.py:1007]: How about just returning m without the slice? Or do you want to do an explicit copy here. If so, clone might be the way to go.

File: heat/core/manipulations.py [hunk 7]
```diff
@@ -936,6 +938,86 @@ def reshape_argsort_counts_displs(
...
+    if k == 2:
+        return flip(flip(m, axes[0]), axes[1])
```
Comment (Markus-Goetz) [inline heat/core/manipulations.py:1009]: Albeit not wrong, highly inefficient in terms of communication. I would think local rotations plus communication would be better.

File: heat/core/manipulations.py (hunk 6 above)
Comment (mtar) [reply heat/core/manipulations.py]: I can't really decide between these options, but a copy is more consistent with the other cases.

File: heat/core/manipulations.py (hunk 7 above)
Comment (mtar) [reply heat/core/manipulations.py:1024]: Can you please elaborate on this.
flip() performs a local operation if split not the axis. Or Is there communication in the creation of an intermediate DNDarray?

File: heat/core/manipulations.py (hunk 6 above)
Comment (Markus-Goetz) [reply heat/core/manipulations.py]: Copy is more consistent

File: heat/core/manipulations.py (hunk 5 above)
Comment (Markus-Goetz) [reply heat/core/manipulations.py:1022]: Nevermind, supposed to be skipped

File: heat/core/manipulations.py (hunk 3 above)
Comment (Markus-Goetz) [reply heat/core/manipulations.py:956]: Nevermind, supposed to be skipped

File: heat/core/manipulations.py (hunk 7 above)
Comment (Markus-Goetz) [reply heat/core/manipulations.py:1027]: I was wrong, either one of the flips is local before/after the one along the split.

**Comment by Markus-Goetz** [comment]
Bump, master needs merge

**Pull request merged**